### PR TITLE
[issue-62] Update Pravega version

### PR DESCRIPTION
**Pull request by vijikarthi**
Signed-off-by: Vijay Srinivasaraghavan <vijayaraghavan.srinivasaraghavan@emc.com>

**Change log description**

  * updated Pravega version to 0.4.0-50.da91e55-SNAPSHOT
  * updated Pravega submodule commit to da91e5518b9797306e94bb383508e59a96f3bc23
  * replaced deprecated BatchClient's getStreamInfo with StreamManager's getStreamInfo
  * fixed failing test cases

**Purpose of the change**
To address https://github.com/pravega/hadoop-connectors/issues/62

**What the code does**
Updates Pravega dependency to latest published artifact.

**How to verify it**
`./gradlew clean build` should pass

### CODE REVIEW

**Review by kevinhan88 — APPROVED** [review]

File: src/test/java/io/pravega/connectors/hadoop/utils/SetupUtils.java [hunk 1]
```diff
@@ -85,11 +85,17 @@ public void startAllServices(String scope) throws Exception {
...
                 .segmentStoreCount(1)
                 .containerCount(4)
+                .enableTls(false)
                 // to match code change https://github.com/pravega/pravega/pull/2476
                 .keyFile("")
                 .certFile("")
+                .enableAuth(false)
                 .userName("")
                 .passwd("")
+                .passwdFile("")
```
Comment (RaulGracia) [inline src/test/java/io/pravega/connectors/hadoop/utils/SetupUtils.java:95]: Why this `SetupUtils` does not populate `inProcPravegaCluster` with real config values? This is different from the FlinkConnector.

File: src/test/java/io/pravega/connectors/hadoop/PravegaInputFormatTest.java [hunk 2]
```diff
@@ -105,6 +107,16 @@ private ClientFactory mockClientFactory() {
         return mockClientFactory;
     }
 
+    private StreamManager mockStreamManager() {
+        StreamManager streamManager = mock(StreamManager.class);
+        io.pravega.client.admin.StreamInfo streamInfo = mock(io.pravega.client.admin.StreamInfo.class);
```
Comment (RaulGracia) [inline src/test/java/io/pravega/connectors/hadoop/PravegaInputFormatTest.java:112]: Hopefully, in the future we may work only with one `StreamInfo` class, so we will not have this import problem.

File: src/test/java/io/pravega/connectors/hadoop/utils/SetupUtils.java (hunk 1 above)
Comment (vijikarthi) [reply src/test/java/io/pravega/connectors/hadoop/utils/SetupUtils.java:95]: Yes. I have found that `PravegaConfig` configurations are not used in Hadoop connectors. Created an issue to address the missing implementation https://github.com/pravega/hadoop-connectors/issues/64

**Comment by RaulGracia** [comment]
@vijikarthi have you executed a Hadoop sample with the connector including these changes?

**Comment by vijikarthi** [comment]
@RaulGracia I have tested `wordcount` sample program with the hadoop connectors artifacts from this PR and it worked. Once this PR is merged, we need to fix samples to address compile errors since the static configurations defined in `PravegaInputFormat` are now moved to `PravegaConfig` class.

**Review by RaulGracia — APPROVED** [review]

**Comment by vijikarthi** [comment]
Can this PR be merged?

**Pull request merged**